R/compat-types-check.R

```diff
@@ -250,6 +255,9 @@ is_number <- function(x,
   if (!typeof(x) %in% c("integer", "double")) {
     return(FALSE)
   }
+  if (!is.numeric(x)) {
```

Comment:
This normally implies that the storage type is int/dbl but better double-check in case of improperly implemented methods.